glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+            path.line_to((400.0, 500.0));
+            path.line_to((500.0, 400.0));
+            path.line_to((0.0, 400.0));
+            path.line_to((100.0, 500.0));
+            path.line_to((100.0, 0.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn doubleCornerGlyphTrickyBitInMiddle() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((100.0, 500.0));
+            path.line_to((100.0, 0.0));
+            path.line_to((400.0, 0.0));
+            path.line_to((400.0, 500.0));
+            path.line_to((500.0, 400.0));
+            path.line_to((0.0, 400.0));
+            path.line_to((100.0, 500.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn curveCorner() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((316.0, 437.0));
+            path.curve_to(
+                (388.67761, 437.0),
+                (446.1305580343, 401.4757887467),
+                (475.0, 344.0),
+            );
+            path.line_to((588.0, 407.0));
+            path.line_to((567.0, 260.0));
+            path.curve_to((567.0, 414.0), (464.0, 510.0), (316.0, 510.0));
+            path.line_to((316.0, 437.0));
+            path.close_path();
+            path
+        }
+
+        pub(super) fn dotabove_ar() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((58.0, -58.0));
+            path.curve_to((90.0, -58.0), (116.0, -32.0), (116.0, 0.0));
+            path.curve_to((116.0, 32.0), (90.0, 58.0), (58.0, 58.0));
+            path.curve_to((26.0, 58.0), (0.0, 32.0), (0.0, 0.0));
+            path.curve_to((0.0, -32.0), (26.0, -58.0), (58.0, -58.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn sofiaSans() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((190.0, 327.0));
+            path.line_to((199.0, 497.0));
+            path.line_to((199.0, 488.0));
+            path.line_to((32.0, 503.0));
+            path.line_to((190.0, 327.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn largeCrossing() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((378.0, 615.0));
+            path.line_to((344.0, 706.0));
+            path.line_to((444.0, 660.0));
+            path.line_to((281.0, 699.0));
+            path.line_to((378.0, 615.0));
+            path.close_path();
+            path
+        }
+    }
+
+    macro_rules! assert_approx {
+        ($left:expr, $right:expr) => {
+            assert_approx!($left, $right, 1e-4)
+        };
+        ($left:expr, $right:expr, $accuracy:literal) => {
+            assert!(
+                ($left - $right).abs() < $accuracy,
+                "{} !~= {}",
+                $left,
+                $right
+            )
+        };
+    }
```

Comment:
I'm failing to grasp why this needs to be a macro? If you made it a free fn that took things that impl a trait you could implement for points which would be nicer than doing x and y as individual asserts.

Reply:
the macro is variadic (has an optional argument) and also generates the panic at the assert site, including the printed message; to achieve this with a fn we would be replacing,

```rust
assert_approx!(new_pt.x, 114.5417);
```
with,
```rust
assert!(approx_eq(new_pt.x, 114.5417, 1e-4), "{new_pt.x} !~= 114.5417");
```

(you could have the assert in the body of the free fn, but then your backtrace originates there, instead of at the call site, which is a small thing but bugs me.)


this could be rewritten to operate on points instead of floats, but i was pretty mindlessly porting the python and these are separate there, so 🤷

Reply:
Hm, I don't think `assert!(approx_eq(new_pt.x, 114.5417, 1e-4), "{new_pt.x} !~= 114.5417");` is on the table, I'd have expected something like `assert_approx_eq(pt, (42, 43));`. I hadn't considered callstack root, that seems worth a comment?